icubam/www/handlers/db.py

```diff
@@ -83,3 +88,36 @@ def get(self, collection):
       os.remove(tmp_path)
     else:
       self.write(data.to_html())
+
+  @tornado.web.authenticated
+  def post(self, collection):
+    if collection == 'bedcounts':
+      csvp = synchronizer.CSVPreprocessor(self.db)
+
+      # Get the file object and format request:
+      file = self.request.files["file"][0]
+      file_format = self.get_query_argument('format', default=None)
+
+      file_name = None
+      # Pre-process with the correct method:
+      if file_format == 'ror_idf':
+        input_buf = io.StringIO(file["body"].decode('utf-8'))
+        csvp.sync_bedcounts_ror_idf(input_buf)
+        file_name = 'ror_idf'
+      else:
+        logging.debug("API called with incorrect file_format: {file_format}.")
+        self.redirect(home.HomeHandler.ROUTE)
+        return
+
+      # Save the file locally just in case:
+      time_str = datetime.now().strftime('%Y-%m-%d-%H:%M:%S')
+      file_path = os.path.join(self.upload_path, f"{time_str}-{file_name}")
+      try:
+        with open(file_path, "wb") as f:
+          f.write(file["body"])
+        logging.info(f"Received {file_path} from {self.request.remote_ip}.")
+      except IOError as e:
+        logging.error(f"Failed to write file due to IOError: {e}")
+    else:
+      logging.debug("POST API called with incorrect collection: {collection}.")
+      self.redirect(home.HomeHandler.ROUTE)
```

Comment:
This should return an HTTP error code. If it is an API handler, I believe you shouldn't even redirect. It will make it easier for the client (whether an UI or test code) to detect and address the error.

Similar remark for other error cases above.

Reply:
Agreed, I won't lie, I wasn't sure how to do that.

Reply:
I agree with Maxime. it is a post request so it should return an error instead of redirecting.
`self.set_status(4xx)` ?

Reply:
Plus I would start with checking it. this would prevent you from indenting a big block:
```
if collection != 'bedcounts':
  logging.error('something')
  self.set_status(400)
  return

...
```

Reply:
I want to first check the endpoint is correct, the block is for routing if we have other POST endpoints.

Reply:
But I've put a global check at the beginning for 404s.